pkg/admin/data/data_manager.go

```diff
@@ -223,7 +224,15 @@ func checkDirectoryContents(existing sets.Set[string]) error {
 }
 
 func copyPath(src, dest string) error {
-	cmd := exec.Command("cp", append(cpArgs, src, dest)...) //nolint:gosec
+	tmpDest := fmt.Sprintf("%s.tmp", dest)
+	if exists, err := pathExists(tmpDest); err != nil {
+		return err
+	} else if exists {
+		if err := os.RemoveAll(tmpDest); err != nil {
```

Comment:
There are quite a few attempts in this function to remove tmpDest data. Is it better handled once at the function exit, as a deferred operation?

Reply:
But they have different purposes at different times.
This one is to remove it if already exists (like it was interrupted etc.) so it needs to happen in this place.
Others are cleaning up in case of a problem and immediately exiting. I don't want to add a defer that "runs wild".

Reply:
I thought about this, but:
- in next PR I plan to refactor this into different way to accommodate "copy all", then "rename all" we'll need for restore
- if we just `defer RemoveAll(tmpDest)`, it'll run even if we don't want it (like the Rename was successful) - as errors should not be ignored, if we log it, it'll be misleading (wanting to remove non-existent file when we don't need to)